extensions/wikia/Search/classes/Services/ESFandomSearchService.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+class ESFandomSearchService extends ESEntitySearchService {
+	const RESULTS_COUNT = 6;
+
+	protected function prepareQuery( $phrase ) {
+		return urlencode( $phrase );
+	}
+
+	protected function select( $query ) {
+		global $wgConsulServiceTag, $wgConsulUrl;
+
+		$consulUrl =
+			( new \Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl,
+				$wgConsulServiceTag ) )->getUrl( 'fandom-search' );
+
+		$response = \Http::post( "http://$consulUrl/fandom", array(
+			'noProxy' => true,
+			'postData' => "queryTerms=$query",
+			'headers' => [
+				'Content-Type' => 'application/x-www-form-urlencoded',
+			],
+		) );
+		if ( $response !== false ) {
+
+			$decodedResponse = json_decode( $response, true );
+			if ( isset( $decodedResponse['hits']['hits'] ) &&
+			     json_last_error() === JSON_ERROR_NONE
+			) {
+				return $decodedResponse['hits']['hits'];
+			}
+		}
+
+		return [];
+	}
+
+	protected function consumeResponse( $response ) {
+		$results = [];
+
+		foreach ( $response as $item ) {
+			if ( isset( $item['_source'] ) ) {
+				$source = $item['_source'];
+
+				$results[] = [
+					'title' => html_entity_decode( $source['title'] ),
+					'excerpt' => isset( $source['excerpt'] )
+						? html_entity_decode( $source['excerpt'] ) : '',
+					'vertical' => html_entity_decode( $source['vertical'] ),
+					'image' => html_entity_decode( $source['image_url'] ),
+					'url' => html_entity_decode( $source['url'] ),
+				];
+				if ( count( $results ) >= ESFandomSearchService::RESULTS_COUNT ) {
+					break;
```

Comment:
With regard to the ```goto```, IMO there's nothing inherently wrong with a ```goto```.  In fact, every loop, every conditional, every switch statement, every function or method has an implied ```goto``` under the hood.  The problem isn't the ```goto```, it's the sloppy code that usually surrounds it.  In this case, however, when we can assume a non-zero slice of data a simple break shouldn't be a big deal, as you say above.

This code will be revisited, I can promise you, because although it's written today as a direct, plug-in replacement of the existing fandom stories search for the sake of expediency it has an inherent flaw which is that it will not support fandom stories search for more than ```RESULTS_COUNT``` items.  For this we need a paging interface that doesn't slice data off any of the returned pages.  This is planned, but as we have much work to do to get discussion search up and running still on our plate for this year we didn't have the luxury of implementing it now.  So the slicing is likely to completely disappear in the next 6 months when we implement our "All" search page with fandom stories as a class of search results.